third_party/nvfuser/csrc/scheduler/normalization.cpp

```diff
@@ -1,35 +1,232 @@
+  rparams->combined_inner_outer = true;
+
+  const auto dev_prop = at::cuda::getCurrentDeviceProperties();
+  const int64_t device_multiprocessor_count =
+      (int64_t)dev_prop->multiProcessorCount;
+
+  // step-1, reduction dim of inner reduction, vect * batch * threads_per_block
+  // >= inner_dim_numel vect represents unroll_factor_inner_reduction
+  int64_t vect = vectorize_factor;
+  int64_t after_vect = ceilDiv(inner_dim_numel, vect);
+
+  // split after_vect between threads_per_block and batch
+  // tuned for layer_norm backward on A100
+  // large batch size leads small block size and low intra-threads
+  // communications with more work per thread small batch size leads large block
+  // size and high intra-threads communications with less work per thread
+  // batch represents batches_per_block_inner_reduction
+  int64_t threads_per_block = 32;
+  int64_t batch = ceilDiv(after_vect, threads_per_block);
+  int64_t batch_max = 20;
+  int64_t batch_min;
+  if (inner_dim_numel >= 3072) {
+    // allow small batch for small iter numbers, e.g. shape= 2048 x 3072
+    if (outer_dim_numel <= 2048 && inner_dim_numel == 3072) {
+      batch_min = 3;
+    } else {
+      batch_min = 4;
+    }
+  } else if (inner_dim_numel >= 2048) {
+    batch_min = 2;
+  } else {
+    batch_min = 1;
+  }
+  // split workload between batch and threads_per_block
+  // decrease batch and increase threads_per_block while current bath is larger
+  // batch_min and current threads_per_block is larger than
+  // threads_per_block_max
+  int64_t threads_per_block_max = inner_dim_numel >= 20480 ? 512 : 256;
+  while (batch > batch_min && threads_per_block < threads_per_block_max) {
+    if (batch % 2 == 0 && batch / 2 >= batch_min) {
+      batch /= 2;
+      threads_per_block *= 2;
+    } else if (batch % 3 == 0 && batch / 3 >= batch_min) {
+      batch /= 3;
+      threads_per_block *= 3;
+    } else if (batch % 5 == 0 && batch / 5 >= batch_min) {
+      batch /= 5;
+      threads_per_block *= 5;
+    } else if (batch > batch_max) {
+      threads_per_block = scheduler_utils::roundUpToN(
+          ceilDiv(after_vect, batch_max), dev_prop->warpSize);
+      batch = ceilDiv(after_vect, threads_per_block);
+      break;
+    } else {
+      break;
+    }
+  }
+  // step-2, iteration dim of outer reduction, (inner_dim_numel)
+  // gdimy*unroll*bdimx >= inner_dim_numel
+  int64_t unroll_factor = inner_dim_numel >= 4096 ? 4 : 2;
+  int64_t after_unroll = ceilDiv(inner_dim_numel, unroll_factor);
+  int64_t blocks_per_sm_max1 =
+      ceilDiv(after_unroll, device_multiprocessor_count);
+
+  // this reg_per_thread is just used to estimate threads_per_sm
+  // the actual maxrregcount is set using getRegPerThreadGivenThreadsPerSM
+  // the estimation is based on layer_norm, the estimated result is larger than
+  // actual register usage and no register spill observed
+  constexpr int64_t overhead_register = 40;
+  constexpr int64_t bytes_per_register = 4;
```

Comment:
nit: sizeof(float)

Also, it's fine to assume `float` but can you add an assertion?